### PR TITLE
fix: set MagicDir to tempdir when performing cache probe

### PR DESCRIPTION
**Pull request by johnstcn**
Relates to https://github.com/coder/terraform-provider-envbuilder/issues/43
Depends on https://github.com/coder/envbuilder/pull/337 and https://github.com/coder/envbuilder/pull/336

- Sets MagicDir to tempDir when performing a cache probe.
- Set up auth for the test registry by default to make sure we are exercising that code path.
- Removes references to RemoteRepoDir (ref: https://github.com/coder/envbuilder/pull/340)

### CODE REVIEW

File: internal/imgutil/imgutil.go [hunk 1]
```diff
@@ -34,7 +33,7 @@ func GetRemoteImage(imgRef string) (v1.Image, error) {
 // ExtractEnvbuilderFromImage reads the image located at imgRef and extracts
 // MagicBinaryLocation to destPath.
 func ExtractEnvbuilderFromImage(ctx context.Context, imgRef, destPath string) error {
-	needle := filepath.Clean(constants.MagicBinaryLocation)[1:] // skip leading '/'
+	needle := ".envbuilder/bin/envbuilder"
```
Comment (dannykopping) [inline internal/imgutil/imgutil.go:36]: Can we use a constant rather for `.envbuilder` in this and `runCacheProbe`?

**Review by dannykopping — APPROVED** [review]

File: internal/imgutil/imgutil.go (hunk 1 above)
Comment (johnstcn) [reply internal/imgutil/imgutil.go:36]: Sure thing -- I was perhaps a bit over-zealous in de-constantizing.

**Review by mtojek — COMMENTED** [review]
Looks like linter is complaining a bit?

**Comment by johnstcn** [comment]
> Looks like linter is complaining a bit?

Can't build until the linked envbuilder PRs are merged :-)

File: internal/imgutil/imgutil.go [hunk 2]
```diff
@@ -34,7 +35,9 @@ func GetRemoteImage(imgRef string) (v1.Image, error) {
 // ExtractEnvbuilderFromImage reads the image located at imgRef and extracts
 // MagicBinaryLocation to destPath.
 func ExtractEnvbuilderFromImage(ctx context.Context, imgRef, destPath string) error {
-	needle := filepath.Clean(constants.MagicBinaryLocation)[1:] // skip leading '/'
+	var o eboptions.Options
+	o.SetDefaults()
```
Comment (mafredri) [inline internal/imgutil/imgutil.go:39]: (envbuilder review: Usage here might be nicer with `o := eboptions.Defaults()`.)

File: internal/provider/cached_image_resource.go [hunk 3]
```diff
@@ -467,6 +466,8 @@ func runCacheProbe(ctx context.Context, builderImage string, opts eboptions.Opti
 	if err := os.MkdirAll(tmpKanikoDir, 0o755); err != nil {
 		return nil, fmt.Errorf("failed to create kaniko dir: %w", err)
 	}
+	// Use the temporary directory as our 'magic dir'.
+	opts.MagicDirBase = tmpKanikoDir
```
Comment (mafredri) [inline internal/provider/cached_image_resource.go:470]: (envbuilder nit: "base" makes me think of the directory that should contain the magicdir, i.e. `[base]/.envbuilder`).

**Review by mafredri — APPROVED** [review]
Just a few envbuilder related comments, this PR is fine 👍🏻

**Pull request merged**